src/pages/[SbtAddress].tsx

```diff
@@ -0,0 +1,282 @@
+import {
+  useAccount,
+  useContract,
+  useStarknetCall,
+  useStarknetExecute,
+  useTransaction,
+} from "@starknet-react/core";
+import { NextPage } from "next";
+import { useRouter } from "next/router";
+import React, { ReactElement, useEffect, useState } from "react";
+import BN from "bn.js";
+import { ec, GetTransactionResponse, hash } from "starknet";
+import Button from "@/components/UI/button";
+import SelectIdentity from "@/components/selectIdentity";
+import Connect from "@/components/connection/connect";
+import styles from "@/styles/generator.module.css";
+import sbt_abi from "@/abi/starknet/sbt_abi.json";
+import { TextField } from "@mui/material";
+import LoadingScreen from "@/components/UI/screens/loadingScreen";
+
+type MetadataProps = {
+  name: string;
+  description: string;
+  image: string;
+};
+
+type CallDataProps = {
+  calldata: (string | number)[];
+  contractAddress: string;
+  entrypoint: string;
+};
+
+type TransactionDatas = {
+  status: string;
+};
+
+const SbtGenerator: NextPage = () => {
```

Comment:
It's a bit weird that the page name is `SbtAddress` and the component itself is named `SbtGenerator` don't you think ? 

to me it should be something like `sbtMintingPage` on both

Reply:
`SbtMintingPage` is much better, I replaced it
You mean also replacing `[SbtAddress]` by `SbtMintingPage` ? the page has a dynamic route which is based on the SBT contractAddress that's why I name it like this